File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANRecordCreateForm.cs
using System;
using System.Windows.Forms;
using Neurotec.Biometrics.Standards;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANRecordCreateForm : Form
	{
		#region Private fields

		private ANTemplate _template;
		private ANRecord _createdRecord;

		#endregion

		#region Public constructor

		public ANRecordCreateForm()
		{
			InitializeComponent();
		}

		#endregion

		#region Public properties

		public ANTemplate Template
		{
			get
			{
				return _template;
			}
			set
			{
				_template = value;
			}
		}

		public ANRecord CreatedRecord
		{
			get
			{
				return _createdRecord;
			}
			protected set
			{
				_createdRecord = value;
			}
		}

		public int Idc
		{
			get
			{
				return (int)nudIdc.Value;
			}
			set
			{
				nudIdc.Value = value;
			}
		}

		#endregion

		#region Protected methods

		protected virtual ANRecord OnCreateRecord(ANTemplate template)
		{
			return null;
		}

		#endregion

		#region Private methods

		private void ANRecordCreateFormFormClosing(object sender, FormClosingEventArgs e)
		{
			if (DialogResult != DialogResult.OK)
			{
				return;
			}

			if (!ValidateChildren())
			{
				e.Cancel = true;
				return;
			}

			try
			{
				CreatedRecord = OnCreateRecord(Template);
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.ToString());
				e.Cancel = true;
			}
		}

		#endregion
	}
}

File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANRecordCreateForm.vb
Imports System
Imports System.Windows.Forms
Imports Neurotec.Biometrics.Standards

Namespace RecordCreateForms

	Partial Public Class ANRecordCreateForm
		Inherits Form
#Region "Private fields"

		Private _template As ANTemplate
		Private _createdRecord As ANRecord

#End Region

#Region "Public constructor"

		Public Sub New()
			InitializeComponent()
		End Sub

#End Region

#Region "Public properties"

		Public Property Template() As ANTemplate
			Get
				Return _template
			End Get
			Set(ByVal value As ANTemplate)
				_template = value
			End Set
		End Property

		Public Property CreatedRecord() As ANRecord
			Get
				Return _createdRecord
			End Get
			Protected Set(ByVal value As ANRecord)
				_createdRecord = value
			End Set
		End Property

		Public Property Idc() As Integer
			Get
				Return Convert.ToInt32(nudIdc.Value)
			End Get
			Set(ByVal value As Integer)
				nudIdc.Value = value
			End Set
		End Property

#End Region

#Region "Protected methods"

		Protected Overridable Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Return Nothing
		End Function

#End Region

#Region "Private methods"

		Private Sub ANRecordCreateFormFormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles MyBase.FormClosing
			If DialogResult <> Windows.Forms.DialogResult.OK Then
				Return
			End If

			If (Not ValidateChildren()) Then
				e.Cancel = True
				Return
			End If

			Try
				CreatedRecord = OnCreateRecord(Template)
			Catch ex As Exception
				MessageBox.Show(ex.ToString())
				e.Cancel = True
			End Try
		End Sub

#End Region
	End Class
End Namespace
